addons/binding/org.openhab.binding.unifi/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,137 @@
+
+	<bridge-type id="controller">
+
+		<label>UniFi Controller</label>
+		<description>A UniFi controller.</description>
+
+		<config-description>
+			<parameter name="host" type="text" required="true">
+				<label>Hostname</label>
+				<description>Hostname of IP address of the UniFi Controller</description>
+				<default>unifi</default>
+				<context>network-address</context>
+			</parameter>
+			<parameter name="port" type="integer" max="65535" min="1" required="true">
+				<label>Port</label>
+				<description>Port of the UniFi Controller</description>
+				<default>8443</default>
+			</parameter>
+			<parameter name="username" type="text" required="true">
+				<label>Username</label>
+				<description>The username to access the UniFi Controller.</description>
+			</parameter>
+			<parameter name="password" type="text" required="true">
+				<label>Password</label>
+				<description>The password to access the UniFi Controller.</description>
+				<context>password</context>
+			</parameter>
+			<parameter name="refresh" type="integer" required="false">
+				<label>Refresh Interval</label>
+				<description>The refresh interval in seconds to poll the UniFi controller</description>
+				<default>10</default>
+			</parameter>
+			<parameter name="considerHome" type="integer" required="false">
+				<label>Consider Home Interval</label>
+				<description>The interval in seconds used to consider a client as home</description>
+				<default>180</default>
+			</parameter>
+		</config-description>
+
+	</bridge-type>
+
+	<thing-type id="client">
+
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="controller" />
+		</supported-bridge-type-refs>
+		<label>UniFi Wireless Client</label>
+		<description>A wireless client connected to a UniFi wireless network</description>
+
+		<channels>
+			<channel id="online" typeId="online"/>
+			<channel id="site" typeId="site" />
+			<channel id="ap" typeId="ap" />
+			<channel id="essid" typeId="essid" />
+			<channel id="rssi" typeId="rssi" />
+			<channel id="uptime" typeId="uptime" />
+			<channel id="lastSeen" typeId="lastSeen" />
+		</channels>
+
+		<config-description>
+			<parameter name="mac" type="text" required="true">
+				<label>MAC Address</label>
+				<description>The MAC address of the client</description>
+			</parameter>
+			<parameter name="site" type="text" required="false">
+				<label>Site</label>
+				<description>The UniFi site where the client should be found (optional)</description>
+			</parameter>
+			<parameter name="contactType" type="text" required="false">
+				<label>Contact Type</label>
+				<description>The contact type for the online channel</description>
+				<options>
+					<option value="NO">Normally Open</option>
+					<option value="NC">Normally Closed</option>
+				</options>
+				<limitToOptions>true</limitToOptions>
+				<default>NO</default>
+			</parameter>
+		</config-description>
+		
+	</thing-type>
+	
+	<channel-type id="online">
+		<item-type>Contact</item-type>
```

Comment:
Contact is pretty unusual here - we usually use Switch, see e.g. the Network binding, which has a similar feature.

Reply:
We talked a little about this in the discussion thread and I opted for Contact since the HOME vs AWAY state is read only. I can change it back to Switch if necessary.

Reply:
Read-only shouldn't be an argument as you can declare that for switches as well - so yes, I would very much prefer to have this changed to Switch.

Reply:
Actually the FritzBox TR64 OH 1.x binding I use at the moment also uses Contacts for online detection...

Reply:
Should be changed as well then - latest when being refactored to a 2.x binding...